### PR TITLE
SDCICD-595: Refactor out synchronizedbeforesuite to a new function

### PR DESCRIPTION
**Pull request by jeefy**
We were getting errors when attempting to run Ginkgo a second time post-Upgrades. Since the use case for SynchronizedBeforeSuite is not needed anymore, we should remove it. 

Ran a full set of upgrade tests locally after this change and things came back solid. 

During debugging, also upgraded `gomega` and `ginkgo` packages.

### CODE REVIEW

File: pkg/common/events/events.go [hunk 1]
```diff
@@ -24,15 +22,14 @@ func initializeEvents() {
 }
 
 // HandleErrorWithEvents returns a gomega assertion and records events depending on the error state.
```
Comment (whereswaldon) [inline pkg/common/events/events.go:24]: Can we update the docs for this function given that we changed the signature?

File: pkg/e2e/e2e.go [hunk 2]
```diff
@@ -81,19 +80,16 @@ var _ = ginkgo.BeforeEach(func() {
 	}
 })
 
-// Setup cluster before testing begins.
-var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
-	defer ginkgo.GinkgoRecover()
-
+func beforeSuite() bool {
```
Comment (whereswaldon) [inline pkg/e2e/e2e.go:83]: Can we document the semantics of the boolean return value?

File: pkg/e2e/e2e.go [hunk 3]
```diff
@@ -241,8 +233,8 @@ func runGinkgoTests() error {
 	gomega.RegisterFailHandler(ginkgo.Fail)
 
 	ginkgoConfig.DefaultReporterConfig.NoisySkippings = !viper.GetBool(config.Tests.SuppressSkipNotifications)
-	ginkgoConfig.GinkgoConfig.SkipString = viper.GetString(config.Tests.GinkgoSkip)
-	ginkgoConfig.GinkgoConfig.FocusString = viper.GetString(config.Tests.GinkgoFocus)
+	ginkgoConfig.GinkgoConfig.SkipStrings = []string{viper.GetString(config.Tests.GinkgoSkip)}
```
Comment (whereswaldon) [inline pkg/e2e/e2e.go:236]: Is this an unrelated bugfix?

File: go.mod [hunk 4]
```diff
@@ -24,8 +24,8 @@ require (
 	github.com/joshdk/go-junit v0.0.0-20201221202203-061ee62ada40
 	github.com/kylelemons/godebug v1.1.0
 	github.com/markbates/pkger v0.17.1
-	github.com/onsi/ginkgo v1.14.1
-	github.com/onsi/gomega v1.10.2
+	github.com/onsi/ginkgo v1.15.0
+	github.com/onsi/gomega v1.11.0
```
Comment (whereswaldon) [inline go.mod:28]: Why update these deps? Was there a feature we needed in there?

**Review by whereswaldon — CHANGES_REQUESTED** [review]
I have a bunch of little questions, but nothing major jumped out.

File: go.mod (hunk 4 above)
Comment (jeefy) [reply go.mod:28]: There wasn't, this will be removed.

File: pkg/e2e/e2e.go (hunk 3 above)
Comment (jeefy) [reply pkg/e2e/e2e.go:236]: This was related to upgrading Ginkgo versions and will be removed.

**Comment by jeefy** [comment]
@whereswaldon PTAL :)

File: pkg/e2e/e2e.go [hunk 5]
```diff
@@ -650,6 +644,13 @@ func runTestsInPhase(phase string, description string, dryrun bool) bool {
 	phaseReporter := ginkgorep.NewPhaseReporter(phase, phaseReportPath)
 	ginkgoPassed := false
 
+	if !dryrun || ginkgoConfig.GinkgoConfig.DryRun {
+		if result := beforeSuite(); result == false {
```
Comment (whereswaldon) [inline pkg/e2e/e2e.go:648]:
```suggestion
		if !beforeSuite() {
```

File: pkg/e2e/e2e.go [hunk 6]
```diff
@@ -650,6 +644,13 @@ func runTestsInPhase(phase string, description string, dryrun bool) bool {
 	phaseReporter := ginkgorep.NewPhaseReporter(phase, phaseReportPath)
 	ginkgoPassed := false
 
+	if !dryrun || ginkgoConfig.GinkgoConfig.DryRun {
```
Comment (whereswaldon) [inline pkg/e2e/e2e.go:647]: So, we only want this to happen if it's *not* a dryrun, right? Shouldn't it be:

```suggestion
	if !dryrun || !ginkgoConfig.GinkgoConfig.DryRun {
```

**Review by whereswaldon — COMMENTED** [review]
/lgtm

**Pull request merged**